bootstrap/postgres/configs.go

```diff
@@ -59,27 +60,33 @@ func (cr configRepository) Save(ctx context.Context, cfg bootstrap.Config, chsCo
 	dbcfg := toDBConfig(cfg)
 
 	if _, err := tx.NamedExec(q, dbcfg); err != nil {
-		e := err
 		if pgErr, ok := err.(*pgconn.PgError); ok && pgErr.Code == pgerrcode.UniqueViolation {
-			e = repoerr.ErrConflict
+			err = repoerr.ErrConflict
 		}
 
-		cr.rollback("Failed to insert a Config", tx)
-		return "", errors.Wrap(repoerr.ErrCreateEntity, e)
+		if errRollback := cr.rollback(err, tx); errRollback != nil {
+			return "", errors.Wrap(err, errRollback)
+		}
 	}
 
 	if err := insertChannels(ctx, cfg.Owner, cfg.Channels, tx); err != nil {
-		cr.rollback("Failed to insert Channels", tx)
+		if errRollback := cr.rollback(err, tx); errRollback != nil {
+			return "", errors.Wrap(err, errRollback)
+		}
 		return "", errors.Wrap(errSaveChannels, err)
 	}
 
 	if err := insertConnections(ctx, cfg, chsConnIDs, tx); err != nil {
-		cr.rollback("Failed to insert connections", tx)
+		if errRollback := cr.rollback(err, tx); errRollback != nil {
+			return "", errors.Wrap(err, errRollback)
+		}
 		return "", errors.Wrap(errSaveConnections, err)
 	}
 
 	if err := tx.Commit(); err != nil {
-		cr.rollback("Failed to commit Config save", tx)
+		if errRollback := cr.rollback(err, tx); errRollback != nil {
+			return "", errors.Wrap(err, errRollback)
+		}
 		return "", err
 	}
 
```

Comment:
Instead having rollback for each step, we can have  defer function like in below example : 

```suggestion
func (cr configRepository) Save(ctx context.Context, cfg bootstrap.Config, chsConnIDs []string) (thingID string, err error) {
	q := `INSERT INTO configs (magistrala_thing, owner, name, client_cert, client_key, ca_cert, magistrala_key, external_id, external_key, content, state)
		  VALUES (:magistrala_thing, :owner, :name, :client_cert, :client_key, :ca_cert, :magistrala_key, :external_id, :external_key, :content, :state)`

	tx, err := cr.db.BeginTxx(ctx, nil)
	if err != nil {
		return "", errors.Wrap(repoerr.ErrCreateEntity, err)
	}
	dbcfg := toDBConfig(cfg)

	defer func(thingID string, err error) (string, error) {
		if err != nil {
			if errRollback := cr.rollback(err, tx); errRollback != nil {
				return thingID, errors.Wrap(err, errRollback)
			}
			return "", err
		}
		return thingID, err
	}(thingID, err)

	if _, err := tx.NamedExec(q, dbcfg); err != nil {
		if pgErr, ok := err.(*pgconn.PgError); ok && pgErr.Code == pgerrcode.UniqueViolation {
			return "", repoerr.ErrConflict
		}
	}

	if err := insertChannels(ctx, cfg.Owner, cfg.Channels, tx); err != nil {
		return "", errors.Wrap(errSaveChannels, err)
	}

	if err := insertConnections(ctx, cfg, chsConnIDs, tx); err != nil {
		return "", errors.Wrap(errSaveConnections, err)
	}

	if err := tx.Commit(); err != nil {
		return "", err
	}
	return cfg.ThingID, nil
}
```